server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/IndexDiskUsageAnalyzer.java

```diff
@@ -11,14 +11,14 @@
 import org.apache.logging.log4j.Logger;
 import org.apache.lucene.backward_codecs.lucene50.Lucene50PostingsFormat;
 import org.apache.lucene.backward_codecs.lucene84.Lucene84PostingsFormat;
+import org.apache.lucene.backward_codecs.lucene90.Lucene90PostingsFormat;
 import org.apache.lucene.codecs.DocValuesProducer;
 import org.apache.lucene.codecs.FieldsProducer;
 import org.apache.lucene.codecs.KnnVectorsReader;
 import org.apache.lucene.codecs.NormsProducer;
 import org.apache.lucene.codecs.PointsReader;
 import org.apache.lucene.codecs.StoredFieldsReader;
 import org.apache.lucene.codecs.TermVectorsReader;
-import org.apache.lucene.codecs.lucene90.Lucene90PostingsFormat;
```

Comment:
I am not sure if besides updating the import for 90 we need an additional conditional for 99 [here](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/IndexDiskUsageAnalyzer.java#L304)

Reply:
So we don't forget, could you make that change in this PR or add a TODO?

Reply:
I pushed the change, I don't feel good about no failing tests with or without it though.

Reply:
> I pushed the change, I don't feel good about no failing tests with or without it though.

No test fails because we miss an inverted index unit test, and the assertion in the IT test is too relaxed. There's a strict assertion in a unit test, but it compares the results between the per-field format and the regular format, both of which silently skip a term state if the format is missing. Can you add an assertion at the end of these format checks to ensure we won't forget to add a new format here? I can add an inverted index unit test next week.

> It seems like we should have one. I honestly don't know, we have one for other older formats :/

Yes, and we should have BWC tests for this API too.

Reply:
Thanks @dnhatn I pushed 120f35cbff534d19f3b8b93bf44b97a5a2d6b670 .